samcli/commands/build/command.py

```diff
@@ -418,3 +433,53 @@ def _process_env_var(container_env_var: Optional[List[str]]) -> Dict:
             processed_env_vars[location_key][env_var_name] = value
 
     return processed_env_vars
+
+
+def _process_image_options(image_args: Optional[Tuple[str]]) -> Dict:
+    """
+    Parameters
+    ----------
+    image_args : Tuple
+        Tuple of command line image options in the format of
+        "Function1=public.ecr.aws/abc/abc:latest" or
+        "public.ecr.aws/abc/abc:latest"
+
+    Returns
+    -------
+    dictionary
+        Function as key and the corresponding image URI as value.
+        Global default image URI is contained in the None key.
+    """
+    build_images: Dict[Optional[str], str] = dict()
+    if image_args:
+        for build_image_string in image_args:
+            function_name, image_uri = _parse_key_value_pair(build_image_string)
+            if not image_uri:
+                LOG.error("Invalid command line --build-image input %s, skipped.", build_image_string)
```

Comment:
Why should we be skipping on invalid input rather than failing the command?

Reply:
Yep I would agree on failing the command here. This behaviour here is to align with how we parse the env vars. But I would agree that the building should not happen if the image cannot be parsed properly

Reply:
Updated.